### DISCUSSION TITLE
Dynamicly inject list

### DISCUSSION THREAD

**Discussion by KiperUladzislau** [dropwizard/dropwizard#4783] [General]
Imagine we have 
```
public interface Provider { 
   String provide();
 }
  ```
and a couple of implementations, for example `ProviderA`, `ProviderB` and `ProviderC`
I want depending on configuration (or whatever) either inject or not some of theese providers via `AbstractBinder`
```
environment.jersey().register(new AbstractBinder() {
            @Override
            protected void configure() {
                if (conditionA) { bind(ProviderA.class).to(Provider.class); }
                if (conditionB) { bind(ProviderB.class).to(Provider.class); }
                if (conditionC) { bind(ProviderC.class).to(Provider.class); }
            }
        });
```
I must bind via `.class` instead of using `new` keyword because dependencies of providers can not be created via `new`
Then what I want is to have somehow only injected providers in some class (for example as `List<Provider>`)
```
public class ProviderServce {
   private List<Provider> providers;
   
   public void doSomething() {
        provider.forEach(...);
   }
  }
```

The problem is that I can not inject such list as dropwizard does not trying to find all implementations of `Provider` and inject them in list (as for example Spring does)
The question is how can I achive such conditional injections with Dropwizard? 
I have an idea to get all implementation of `Provider` from context in which all injected classes are considered to be stored, but I've not found any public method or class which can be customized.
Can you please provide me with some information?
 Thanks in advance

**Comment by joschi** [comment]
@KiperUladzislau The dependency injection framework used in Jersey (by default) is HK2.

Take a look at [IterableProvider injection](https://javaee.github.io/hk2/introduction.html#iterableprovider-injection) for details how to inject multiple implementations of an interface/contract.
